src/commonMain/kotlin/fr/acinq/eclair/io/Peer.kt

```diff
@@ -405,6 +407,30 @@ class Peer(
                     val htlcInfos = db.channels.listHtlcInfos(actualChannelId, action.commitmentNumber).map { ChannelAction.Storage.HtlcInfo(actualChannelId, action.commitmentNumber, it.first, it.second) }
                     input.send(WrappedChannelEvent(actualChannelId, ChannelEvent.GetHtlcInfosResponse(action.revokedCommitTxId, htlcInfos)))
                 }
+                action is ChannelAction.Storage.StoreChannelClosing -> {
+                    val dbId = UUID.fromBytes(channelId.take(16).toByteArray())
+                    val payment = OutgoingPayment(
+                        id = dbId,
+                        recipientAmount = action.amount,
+                        recipient = PublicKey.Generator,
+                        details = OutgoingPayment.Details.ChannelClosing(
+                            channelId = channelId,
+                            closingAddress = action.closingAddress,
+                            isSentToMyWallet = action.isSentToMyWallet,
+                            paymentHash = randomBytes32()
```

Comment:
We could do like we did for swaps, `channelId.sha256()` so that it's deterministic

Reply:
Done in d105365